src/TaskRunner/Commands/ComponentCheckCommands.php

```diff
@@ -221,6 +221,27 @@ protected function componentComposer()
                 }
             }
         }
+
+        // Make sure that the forbidden/obsolete script is not present in the 'scripts' section of composer.json file.
+        if (isset($composerJson['scripts'])) {
+            // Get forbidden/obsolete scripts from config.
+            $scripts = $this->getConfig()->get('toolkit.components.composer.scripts.forbidden');
+            // Define common error message.
+            $error = 'The obsolete invocation of %s is present in composer.json %s scripts. Please remove.';
+            // Detect forbidden scripts in composer.json.
+            foreach ($scripts as $level => $scripts) {
+                foreach ($scripts as $script) {
+                    if (
+                        isset($composerJson['scripts'][$level])
+                        && in_array($script, $composerJson['scripts'][$level])
+                    ) {
```

Comment:
Hi @ozyrys ,

Something is strange here, the same `$scripts` is used in both `foreach`'s.

There's also another use-case that we should consider, when the scripts are defined as a single script instead of an array of scripts. Currently is throwing error.

We use that approach in our composer.json https://github.com/ec-europa/toolkit/blob/release/10.x/composer.json#L64 file.
Another example, the project https://github.com/ec-europa/just-ejustice-reference/blob/master/composer.json#L160

Please have a look to my suggestion.

```suggestion
        if (!empty($composerJson['scripts'])) {
            // Get forbidden/obsolete scripts from config.
            $forbiddenScripts = $this->getConfig()->get('toolkit.components.composer.scripts.forbidden');
            // Define common error message.
            $error = 'The obsolete invocation of %s is present in composer.json %s scripts. Please remove.';
            // Detect forbidden scripts in composer.json.
            foreach ($forbiddenScripts as $level => $scripts) {
                if (!isset($composerJson['scripts'][$level])) {
                    continue;
                }
                foreach ((array) $composerJson['scripts'][$level] as $script) {
                    if (in_array($script, $scripts)) {
```

Reply:
Hi @joaocsilva this has been fixed, please proceed with the review.